### PR TITLE
wait to parser

### PR DESCRIPTION
**Pull request by moshe-kabala**
- Add spiner package
- Add wait code integration status function
- Add function that print the parser error
- Use these function in code and project push

### CODE REVIEW

**Review by asafyehezkel — APPROVED** [review]
LGTM
Please see my comments

File: pkg/code/api.go [hunk 1]
```diff
@@ -117,5 +125,51 @@ func AddCodeIntegrationVersion(ctx context.Context, tarGzFile io.Reader, codeInt
 
 	log.Info("Done!")
 	return res.Dataset.LatestVersion, nil
+}
+
+// WaitForCodeIntegrationStatus waits for the code integration status to reach a desired state.
+// It returns true if the status becomes 'TESTSTATUS_TEST_SUCCESS', false if it becomes 'TESTSTATUS_TEST_FAIL',
+// and an error if any other error occurs or if the waiting times out.
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/code/api.go]: done

File: pkg/code/api.go [hunk 2]
```diff
@@ -117,5 +125,51 @@ func AddCodeIntegrationVersion(ctx context.Context, tarGzFile io.Reader, codeInt
...
+func WaitForCodeIntegrationStatus(ctx context.Context, id string) (ok bool, codeIntegrationVersion *CodeIntegrationVersion, err error) {
+	timeoutDuration := 10 * time.Minute
+
+	// Get the current time
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/code/api.go]: done

File: pkg/code/api.go [hunk 3]
```diff
@@ -117,5 +125,47 @@ func AddCodeIntegrationVersion(ctx context.Context, tarGzFile io.Reader, codeInt
...
+	s := spinner.New(spinner.CharSets[36], 500*time.Millisecond)
+	s.Prefix = "Waiting for code parser result "
+	s.Start()
+
+	defer func() {
+		s.Stop()
+		fmt.Println() // Move to the next line after stopping the spinner
+	}()
+
+	for time.Since(startTime) < timeoutDuration {
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply pkg/code/api.go]: done - I added function called `WaitForCondition`

File: cmd/code/push.go [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
 	"github.com/spf13/cobra"
 	"github.com/tensorleap/leap-cli/pkg/auth"
 	"github.com/tensorleap/leap-cli/pkg/code"
+	"github.com/tensorleap/leap-cli/pkg/log"
 	"github.com/tensorleap/leap-cli/pkg/workspace"
 )
 
 func NewPushCmd() *cobra.Command {
 	var secretId string
+	var watch bool
```
[the remark this answers is not in the document]
Comment (moshe-kabala) [reply cmd/code/push.go:15]: boolean by default is false

**Pull request merged**